### PR TITLE
PR for Pandas issue #14872 / fillna() TZ datetime64 rounded

### LINKED ISSUE #14872: fillna() containing timezone aware datetime64 rounded
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import datetime, pytz

#Non-timezone aware datetime
s = pd.Series( [pd.NaT, pd.NaT, datetime.datetime(2016, 12, 12, 22, 24, 6, 100001) ] )
print(s)
0                          NaT
1                          NaT
2   2016-12-12 22:24:06.100001
dtype: datetime64[ns]

print(s.fillna(method='bfill'))
0   2016-12-12 22:24:06.100001
1   2016-12-12 22:24:06.100001
2   2016-12-12 22:24:06.100001
dtype: datetime64[ns]


# Timezone aware datetime
s = pd.Series( [pd.NaT, pd.NaT, datetime.datetime(2016, 12, 12, 22, 24, 6, 100001, tzinfo=pytz.utc) ] )
print(s)
0                                NaT
1                                NaT
2   2016-12-12 22:24:06.100001+00:00

print(s.fillna(method='bfill'))
0   2016-12-12 22:24:06.100001024+00:00
1   2016-12-12 22:24:06.100001024+00:00
2   2016-12-12 22:24:06.100001024+00:00
dtype: datetime64[ns, UTC]
```

#### Problem description

This is similar to bug #6587 but difference is a timezone aware datetime.  It looks like the dtype is being stripped and the value is converted to a float and back.

#### Expected Output
```python
s = pd.Series( [pd.NaT, pd.NaT, datetime.datetime(2016, 12, 12, 22, 24, 6, 100001, tzinfo=pytz.utc) ] )
print(s)
0                                NaT
1                                NaT
2   2016-12-12 22:24:06.100001+00:00

print(s.fillna(method='bfill'))
0   2016-12-12 22:24:06.100001000+00:00
1   2016-12-12 22:24:06.100001000+00:00
2   2016-12-12 22:24:06.100001000+00:00
dtype: datetime64[ns, UTC]
```

#### Output of ``pd.show_versions()``

<details>
pandas: 0.19.1
numpy: 1.11.2
scipy: 0.18.1
pytz: 2016.7
</details>

### PR DESCRIPTION
**Pull request by RodolfoRFR**
Hi Jeff,

The change adds a "datetime64[ns, UTC]" object to the _DATELIKE_DTYPES set in pandas/types/common.py. 

No such object existed in this set and that is the reason the wrong method was executed on the inputs.

Object is created with "DatetimeTZDtype.__new__"

Thanks,
Rodolfo

 - [X] closes #14872
 - [ X] tests added / passed
 - [ X] passes ``git diff upstream/master | flake8 --diff`` (returns the same number of skips, errors and failures before changes)
 - [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
this goes way overboard 

see the pull request which closed the original issue

**Comment by RodolfoRFR** [comment]
The changes in request #6587  added "dtype in _DATELIKE_DTYPES" to the method selection in the backfill_2d function (now in missing.py), 
```
-        _method = getattr(algos, 'backfill_2d_inplace_%s' % dtype, None)
-    elif is_datetime64_dtype(values):
+        _method = getattr(algos, 'backfill_2d_inplace_%s' % dtype.name, None)
+    elif dtype in _DATELIKE_DTYPES or is_datetime64_dtype(values):
```
I initially tried adding an extra is_datetime64tz_dtype() to this line with no success. After tracing the process with pdb I found that the values are already coerced to integers in pandas/core/internals.py,

```
    def _interpolate_with_fill(self, method='pad', axis=0, inplace=False,
                               limit=None, fill_value=None, coerce=False,
                               downcast=None, mgr=None):
        """ fillna but using the interpolate machinery """
        .
        .
        .
        values, _, fill_value, _ = self._try_coerce_args(values, fill_value)
```
before being passed to /pandas/missing/missing.interpolate_2d and then to backfill_2d. The dtype information is passed as dtype=datetime64[ns, UTC]. When it gets to these lines in the backfill_2d function,
```
elif dtype in _DATELIKE_DTYPES or is_datetime64_dtype(values):
      method = _backfill_2d_datetime
```
the return is always false because datetime64[ns, UTC] is not in _DATELIKE_DTYPES and the 'values' are integers.

Another way would be to either replace datetime64[ns, UTC] with datetime64[ns] before we get to this line or skip the coercion somehow. Unless I am missing something.

**Comment by jreback** [comment]
so this should simply be changed to pass the actual value (and not coerced ones) when this is called and drop al of the _DATELIKE_DTYPES that is really old code

use the is_datetime64_dtype and is_datetime64tz_dtype

**Comment by jreback** [comment]
actually you just need to check the dtype instead (of the values)

e.g.
```
(Pdb) p is_datetime64tz_dtype(dtype)
True
```

**Comment by RodolfoRFR** [comment]
Done! I was indeed missing something. Thank you for the feedback!

File: pandas/tests/test_dtype_utc.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+def test_dtype_utc():
+
+    data = pd.Series([pd.NaT, pd.NaT,
+                      datetime.datetime(2016, 12, 12, 22, 24, 6, 100001,
+                                        tzinfo=pytz.utc)])
+
+    filled = data.fillna(method='bfill')
+
+    expected = pd.Series([datetime.datetime(2016, 12, 12, 22, 24, 6, 100001,
```
Comment (jreback) [inline pandas/tests/test_dtype_utc.py]: this test needs to go in test_missing (search and out it with other similar tests)

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew entry (0.19.2) ok if u can make these changes in next day or so

File: pandas/tests/series/test_missing.py [hunk 2]
```diff
@@ -908,6 +911,24 @@ def test_interp_timedelta64(self):
                           index=pd.to_timedelta([1, 2, 4]))
         assert_series_equal(result, expected)
 
+    # GH 14872
+    def test_dtype_utc():
```
Comment (jreback) [inline pandas/tests/series/test_missing.py]: needs to take self

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**